tests/pos/i18558.scala

```diff
@@ -0,0 +1,16 @@
+
+trait Food[+A]
+trait Animal
+object Animal {
+  extension (animal: Animal) def consume[A](weights: Food[A]): A = ???
+}
+
+val animal: Animal = ???
+val choices: List[Food[Int]] = ???
+val result0: List[Any] = choices.map(animal.consume) // compiles
+// val result1: List[Int] = choices.map(animal.consume) // does not compile
```

Comment:
Shouldn't `result1` compile too ? It does look like it is the expected behaviour and the purpose of the ticket.

Reply:
Yes indeed, I went a little to quickly over that one